lib/core/base/virtual-node/virtual-node.js

```diff
@@ -27,11 +26,7 @@ class VirtualNode extends AbstractVirtualNode {
 
     this._isHidden = null; // will be populated by axe.utils.isHidden
     this._cache = {};
-
-    if (typeof isXHTMLGlobal === 'undefined') {
-      isXHTMLGlobal = isXHTML(node.ownerDocument);
-    }
-    this._isXHTML = isXHTMLGlobal;
+    this._isXHTML = isXHTML(node.ownerDocument);
```

Comment:
I'm not sure we should make this change this way. Can the doc change between XHTML and HTML on the same page? If not then we shouldn't call this for every node as that will create _n_ anchor tags that we have to check the `localName` of. Would rather we only called this function once for creating the entire tree rather than once per node.